rllib/evaluation/env_runner_v2.py

```diff
@@ -173,7 +174,27 @@ def _build_multi_agent_batch(
                 )
             )
 
-        ma_batch[pid] = collector.build()
+        batch = collector.build()
+
+        policy = collector.policy
+
+        if policy.config.get("_enable_rl_module_api", False):
+            # Before we send the collected batch back for training, we may need
+            # to add a time dimension for the RLModule.
+            seq_lens = batch.get(SampleBatch.SEQ_LENS)
+            pad_batch_to_sequences_of_same_size(
+                batch=batch,
+                max_seq_len=policy.config["model"]["max_seq_len"],
+                shuffle=False,
+                batch_divisibility_req=policy.batch_divisibility_req,
+                view_requirements=policy.view_requirements,
+                _enable_rl_module_api=True,
+            )
+            batch = policy.maybe_add_time_dimension(
+                batch, seq_lens=seq_lens, framework="np"
+            )
+
+        ma_batch[pid] = batch
```

Comment:
This is the part where, after finishing collecting a batch, but before sending it back to the Learner, we want to pad it and add a time dimensions already. This is because we don't want the learner to contain this logic.
We can later decide on where exactly this transformation should be when we bring in our new sampling backend. For now, the learner should already expect batches that are "ready".